### DISCUSSION TITLE
Error when upgrading from v1.9.3 (with v2 future flags) to v2: `TypeError: import_node.Response is not a constructor`

### DISCUSSION THREAD

**Discussion by oscard0m** [remix-run/remix#7447] [Q&A] [answered]
I was expecting to upgrade from `v1.9.3 + future flags` to `v2` and have 0 problems but something seems to not be working.

I get the following error:
```
[remix-serve] http://localhost:3000 (http://192.168.241.2:3000)
TypeError: import_node.Response is not a constructor
    at onShellReady (/workspace/app/entry.server.tsx:93:13)
    at finishedTask (/workspace/node_modules/react-dom/cjs/react-dom-server.node.development.js:6513:7)
    at retryTask (/workspace/node_modules/react-dom/cjs/rehttps://codesandbox.io/p/sandbox/remix-v2-forked-dx97sxact-dom-server.node.development.js:6598:5)
    at performWork (/workspace/node_modules/react-dom/cjs/react-dom-server.node.development.js:6642:7)
    at Immediate._onImmediate (/workspace/node_modules/react-dom/cjs/react-dom-server.node.development.js:6969:12)
```

- Example CodeSandbox with Remix v1.9.3: https://codesandbox.io/p/sandbox/remix-v1-with-v2-routing-glcc2w?file=%2F.codesandbox%2FDockerfile%3A1%2C13
- Example CodeSandbox with Remix v2: https://codesandbox.io/p/sandbox/remix-v2-forked-dx97sx

Is there a breaking change with `import { Response } from @remix-run/node`? ~I will try to take a deeper look into this and update the thread with more info~ Has this something to do with [Node built-in polyfills](https://remix.run/docs/en/main/start/v2#servernodebuiltinspolyfill)?

**Comment by oscard0m** [comment, marked as answer]
Checking the templates I'm going to apply this approach + removing the `import { Response } from @remix-run/node` so the native `Response` enters in action:

https://github.com/remix-run/remix/blob/7f1cc39d47c881787556233f0181bd90bf9d0581/templates/remix/app/entry.server.tsx#L106-L116

**Reply by hakimLyon** [reply]
From my side, I imported Response like this using the import statement `import { Response } from "@remix-run/web-fetch"`.😅

**Reply by MichaelDeBoey** [reply]
You should indeed use the native `Response`

If you're not using `remix-serve`, it's recommended to call `installGlobals`

**Comment by Darshan562** [comment]
Your answer is very useful.
